### PR TITLE
detect-pcre/v2: add tests

### PR DESCRIPTION
**Pull request by TheKharleeci**
Test conversion: 
detect-pcre-01: DetectPcreModifPTest04
detect-pcre-02: DetectPcreModifPTest05
detect-pcre-03: DetectPcreTestSig01-03
detect-pcre-04: DetectPcreTestSig09-14
detect-pcre-05: DetectPcreTestSig15-16, DetectPcreFlowvarCapture01- 03

Changes from last PR: added test for DetectPcreFlowvarCapture01- 03

Previous PR: #634

### CODE REVIEW

**Comment by TheKharleeci** [comment]
Corresponding Suricata PR: https://github.com/OISF/suricata/pull/6942

**Review by inashivb — CHANGES_REQUESTED** [review]
There are still some tweaks that we need to make. Comments inline. :)

File: tests/detect-pcre/detect-pcre-02/test.rules [hunk 1]
```diff
@@ -0,0 +1,2 @@
+alert http any any -> any any (msg:"Pcre modifier P"; pcre:"/DOC/P"; sid:1;)
+alert http any any -> any any (msg:"Pcre modifier P (no match)"; pcre:"/DOCTYPE/P"; sid:2;)
```
[the remark this answers is not in the document]
Comment (jufajardini) [reply tests/detect-pcre/detect-pcre-02/test.rules:2]: From the original test, both rules should not alert with the first packet, but should be triggered with the second. Is there a good way to test that exact behavior in a single test, here?

File: tests/detect-pcre/detect-pcre-04/writepcap.py [hunk 2]
```diff
@@ -0,0 +1,9 @@
+#!/usr/bin/env python
+from scapy.all import *
+
+pkts = []
+
+pkts += Ether(dst='ff:ff:ff:ff:ff:ff', src='00:01:02:03:04:05')/ \
+    Dot1Q(vlan=6)/ \
+    IP(dst='1.2.3.4', src='5.6.7.8')/TCP(sport=6666, dport=80, flags='P''A')/"POST / HTTP/1.0\r\nUser-Agent: Mozilla/1.0\r\nCookie: dummy 1234\r\n\r\n"
+wrpcap('input.pcap', pkts)
```
Comment (jufajardini) [inline tests/detect-pcre/detect-pcre-04/writepcap.py:9]: Considering the unittest `DetectPcreTestSig12` I think we should have another packet here, with payload
```c
"GET / HTTP/1.0\r\nUser-Agent: Mozilla/1.0\r\n"
        "Cookie: dummoOOooooO\r\n\r\n";
 ```
for this S-V test to truthfully cover what we had there.

File: tests/detect-pcre/detect-pcre-04/test.rules [hunk 3]
```diff
@@ -0,0 +1,6 @@
+alert http any any -> any any (msg:"HTTP cookie"; pcre:"/dummy/C"; sid:1;)
+alert http any any -> any any (msg:"HTTP method"; pcre:"/POST/M"; sid:2;)
+alert http any any -> any any (msg:"HTTP method"; pcre:!"/POST/M"; sid:3;)
+alert http any any -> any any (msg:"HTTP header"; pcre:"/User[-_]Agent[:]?\sMozilla/H"; sid:4;)
+alert http any any -> any any (msg:"HTTP header"; pcre:!"/User[-_]Agent[:]?\sIEXPLORER/H"; sid:5;)
+alert tcp any any -> any any (content:"POST"; http_method; content:"Mozilla"; http_header; content:"dummy"; http_cookie; pcre:"/one/P"; sid:6; rev:1;)
```
[the remark this answers is not in the document]
Comment (jufajardini) [reply tests/detect-pcre/detect-pcre-04/test.rules:6]: I think this last one was actually taken from `DetectPcreTxBodyChunksTest`s Maybe it wasn't meant to be here?

File: tests/detect-pcre/detect-pcre-05/test.rules [hunk 4]
```diff
@@ -0,0 +1,6 @@
+alert http any any -> any any (msg:"pcre relative HTTP cookie"; content:"dummy"; http_cookie; pcre:"/1234/RC"; sid:1;)
+alert http any any -> any any (msg:"pcre relative HTTP method"; content:"PO"; http_method; pcre:"/ST/RM"; sid:2;)
+alert http any any -> any any (content:"User-Agent: "; http_header; pcre:"/(?P<flow_ua>.*)\r\n/HR"; priority:1; sid:3;)
+alert http any any -> any any (content:"Server: "; http_header; pcre:"/(?P<flow_ua>.*)\r\n/HR"; priority:3; sid:4;)
+alert http any any -> any any (content:"User-Agent: "; http_header; pcre:"/(?P<flow_ua>.*)\r\n/HR"; content:"xyz"; http_header; priority:1; sid:5;)
+alert http any any -> any any (content:"Server: "; http_header; pcre:"/(?P<flow_ua>.*)\r\n/HR"; content:"xyz"; http_header; priority:3; sid:6;)
```
Comment (jufajardini) [inline tests/detect-pcre/detect-pcre-05/test.rules:6]: Is there a preferred way for checking flowvar related sids in a S-V test? I noticed that the corresponding unittests in this case have other checks than an alert check doesn't seem to convey.

File: tests/detect-pcre/detect-pcre-05/writepcap.py [hunk 5]
```diff
@@ -0,0 +1,9 @@
+#!/usr/bin/env python
+from scapy.all import *
+
+pkts = []
+
+pkts += Ether(dst='ff:ff:ff:ff:ff:ff', src='00:01:02:03:04:05')/ \
+    Dot1Q(vlan=6)/ \
+    IP(dst='1.2.3.4', src='5.6.7.8')/TCP(sport=6666, dport=80, flags='P''A')/"POST / HTTP/1.0\r\nUser-Agent: Mozilla/1.0\r\nCookie: dummy 1234\r\n\r\n"
```
Comment (jufajardini) [inline tests/detect-pcre/detect-pcre-05/writepcap.py:8]: The `DetectPcreFlowvarCapture` unittests have a more complete payload, and I noticed that some of the signatures in them do look for content that shows up later on in that payload (`Server`). Is it ok to miss that here?

**Review by jufajardini — CHANGES_REQUESTED** [review]
Some comments inline. I think we need input from another mentor for some of those!

File: tests/detect-pcre/detect-pcre-02/test.rules (hunk 1 above)
[the remark this answers is not in the document]
Comment (inashivb) [reply tests/detect-pcre/detect-pcre-02/test.rules:2]: With what I see here btw I think if fragmentation is done correctly, both the rules should alert.

**Comment by TheKharleeci** [comment]
Followed By: #776

**Pull request closed without merging**